test/remote/gateways/remote_d_local_test.rb

```diff
@@ -268,4 +268,32 @@ def test_transcript_scrubbing
     assert_scrubbed(@credit_card.verification_value, transcript)
     assert_scrubbed(@gateway.options[:trans_key], transcript)
   end
+
+  def test_successful_authorize_with_3ds_v1_options
+    @options[:three_d_secure] = {
+      version: '1.0',
+      cavv: '3q2+78r+ur7erb7vyv66vv\/\/\/\/8=',
+      eci: '05',
+      xid: 'ODUzNTYzOTcwODU5NzY3Qw==',
+      enrolled: 'Y',
+      authentication_response_status: 'Y'
+    }
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+    assert_match 'The payment was authorized', auth.message
+  end
+
+  def test_successful_authorize_with_3ds_v2_options
+    @options[:three_d_secure] = {
+      version: '2.2.0',
+      cavv: '3q2+78r+ur7erb7vyv66vv\/\/\/\/8=',
+      eci: '05',
+      ds_transaction_id: 'ODUzNTYzOTcwODU5NzY3Qw==',
+      enrolled: 'Y',
+      authentication_response_status: 'Y'
+    }
+    auth = @gateway.authorize(@amount, @credit_card, @options)
+    assert_success auth
+    assert_match 'The payment was authorized', auth.message
+  end
```

Comment:
❓ Question:
Did you considered to add an unsuccessful test case?

Reply:
:fu: yes agree, what culd be happen if you use an invalid ds_transaction_id ? this value can be a random string ?

Reply:
( @javierpedrozaing / @gasb150 ) Yup I considered that but there is no way to make that transaction fail because, seems to be the case that for sandbox test there are no checks in the info, and I'm not sure but I think that this info is just to store the approval info the gateway is not going to check if that is true, the 3DsServer already did it.